test/functional/mempool_reorg.py

```diff
@@ -106,8 +106,45 @@ def run_test(self):
 
         self.log.info("Check that the mempool is empty")
         assert_equal(set(self.nodes[0].getrawmempool()), set())
+        assert_equal(set(self.nodes[1].getrawmempool()), set())
         self.sync_all()
 
 
+        self.log.info("Simulate natural reorg")
+        self.connect_nodes(0, 1)
```

Comment:
nit: Why is this needed? Are they not connected already?

Reply:
They don't stay in sync if I don't call `connect_nodes` here... it also doesn't work if I call `connect_nodes(1, 0)` instead. So my guess is that this creation of a manual connection from node0 to node1 adds a permission that we need.

Reply:
Ah ok, it is not possible to use compact blocks during IBD

Reply:
..